Unit test TestDeleteAllProfiles fails: delete_test.go:203: ListProfiles length = 9, expected 8

TestDeleteAllProfiles fails but only after running:
"minikube start"

If "minikube delete" is run - test passes.  

**Steps to reproduce the issue:** 
1. make 
2.  out/minikube start
3. make test

**Full output of failed command:**
[opc@kubernetes1 minikube]$ make test
MINIKUBE_LDFLAGS="-X k8s.io/minikube/pkg/version.version=v1.15.1 -X k8s.io/minikube/pkg/version.isoVersion=v1.15.2-snapshot2 -X k8s.io/minikube/pkg/version.isoPath=minikube/iso -X k8s.io/minikube/pkg/version.gitCommitID="322e138f6bb1c47d7768efe58d62b18e4f4e54d4-dirty" -X k8s.io/minikube/pkg/version.storageProvisionerVersion=v3" ./test.sh 
= make lint =============================================================
ok
= go mod ================================================================
ok
= boilerplate ===========================================================
ok
= schema_check ==========================================================
ok
= go test ===============================================================
--- FAIL: TestDeleteAllProfiles (0.13s)
    delete_test.go:203: ListProfiles length = 9, expected 8
        valid: []
        invalid: [0xc000e3aab0 0xc000e3aae0 0xc000e3ab10 0xc000e3ab40 0xc000e3ab70 0xc000e3aba0 0xc000e3abd0 0xc000e3ac00 0xc000e3ac60]
E1128 10:50:03.679291   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=127.0.0.1:3128 to docker env.
E1128 10:50:03.679984   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=127.0.0.1:3128 to docker env.
E1128 10:50:03.680098   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=localhost:3128 to docker env.
E1128 10:50:03.680275   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=localhost:3128 to docker env.
E1128 10:50:03.680388   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=http://localhost:3128 to docker env.
E1128 10:50:03.680593   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=http://localhost:3128 to docker env.
E1128 10:50:03.680691   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=http://127.0.0.1:3128 to docker env.
E1128 10:50:03.680867   33647 out.go:142] [unset errFile]: ! Local proxy ignored: not passing HTTP_PROXY=http://127.0.0.1:3128 to docker env.
FAIL
coverage: 18.7% of statements
FAIL	k8s.io/minikube/cmd/minikube/cmd	4.819s
ok  	k8s.io/minikube/cmd/minikube/cmd/config	0.084s	coverage: 21.0% of statements
ok  	k8s.io/minikube/pkg/addons	0.113s	coverage: 50.9% of statements
ok  	k8s.io/minikube/pkg/drivers	0.016s	coverage: 19.6% of statements
ok  	k8s.io/minikube/pkg/drivers/hyperkit	0.010s	coverage: 77.3% of statements
ok  	k8s.io/minikube/pkg/drivers/kic/oci	0.018s	coverage: 10.3% of statements
ok  	k8s.io/minikube/pkg/drivers/kvm	0.017s	coverage: 2.1% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper	0.726s	coverage: 55.1% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/bsutil	0.137s	coverage: 62.4% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/bsutil/ktmpl	0.012s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/bootstrapper/images	0.004s	coverage: 82.0% of statements
ok  	k8s.io/minikube/pkg/minikube/cluster	0.082s	coverage: 14.5% of statements
ok  	k8s.io/minikube/pkg/minikube/command	0.061s	coverage: 4.2% of statements
ok  	k8s.io/minikube/pkg/minikube/config	0.104s	coverage: 72.7% of statements
ok  	k8s.io/minikube/pkg/minikube/cruntime	0.078s	coverage: 36.5% of statements
ok  	k8s.io/minikube/pkg/minikube/docker	0.054s	coverage: 20.8% of statements
ok  	k8s.io/minikube/pkg/minikube/driver	0.050s	coverage: 38.3% of statements
ok  	k8s.io/minikube/pkg/minikube/extract	0.035s	coverage: 58.6% of statements
ok  	k8s.io/minikube/pkg/minikube/image	0.048s	coverage: 3.0% of statements
ok  	k8s.io/minikube/pkg/minikube/kubeconfig	0.075s	coverage: 81.3% of statements
ok  	k8s.io/minikube/pkg/minikube/localpath	0.024s	coverage: 48.6% of statements
ok  	k8s.io/minikube/pkg/minikube/logs	0.105s	coverage: 1.3% of statements
ok  	k8s.io/minikube/pkg/minikube/machine	1.135s	coverage: 35.8% of statements
ok  	k8s.io/minikube/pkg/minikube/mustload	0.066s	coverage: 10.5% of statements
ok  	k8s.io/minikube/pkg/minikube/notify	0.054s	coverage: 92.9% of statements
ok  	k8s.io/minikube/pkg/minikube/out	0.021s	coverage: 66.7% of statements
ok  	k8s.io/minikube/pkg/minikube/out/register	0.026s	coverage: 55.4% of statements
ok  	k8s.io/minikube/pkg/minikube/perf	4.051s	coverage: 21.1% of statements
ok  	k8s.io/minikube/pkg/minikube/proxy	0.015s	coverage: 68.7% of statements
ok  	k8s.io/minikube/pkg/minikube/reason	0.034s	coverage: 80.0% of statements
ok  	k8s.io/minikube/pkg/minikube/registry	0.026s	coverage: 76.7% of statements
ok  	k8s.io/minikube/pkg/minikube/service	0.105s	coverage: 84.2% of statements
ok  	k8s.io/minikube/pkg/minikube/shell	0.007s	coverage: 94.3% of statements
ok  	k8s.io/minikube/pkg/minikube/storageclass	0.025s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/style	0.011s	coverage: 100.0% of statements
ok  	k8s.io/minikube/pkg/minikube/sysinit	0.057s	coverage: 6.0% of statements
ok  	k8s.io/minikube/pkg/minikube/translate	0.021s	coverage: 11.4% of statements
ok  	k8s.io/minikube/pkg/minikube/tunnel	1.766s	coverage: 64.2% of statements
ok  	k8s.io/minikube/pkg/util	0.610s	coverage: 75.7% of statements
ok  	k8s.io/minikube/pkg/util/lock	0.021s	coverage: 22.2% of statements
ok  	k8s.io/minikube/pkg/util/retry	0.003s	coverage: 0.0% of statements
FAIL
make: *** [test] Error 32

**Full output of `minikube start` command used, if not already included:**
[opc@kubernetes1 minikube]$ out/minikube start
😄  minikube v1.15.1 on Oracle 7.9
✨  Automatically selected the docker driver
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=3800MB) ...
🐳  Preparing Kubernetes v1.19.4 on Docker 19.03.13 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
💡  kubectl not found. If you need it, try: 'minikube kubectl -- get pods -A'
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default


**My findings**

<details>

TestDeleteAllProfiles expects fixed number of 8 profiles which are taken from test data directory: "../../../pkg/minikube/config/testdata/delete-all".
However, extra profile "minikube" is added in ListProfiles method:
file:  pkg/minikube/config/profile.go

fragment responsible for adding:
```
// try to get profiles list based on all containers created by docker driver
	cs, err := oci.ListOwnedContainers(oci.Docker)
```
</details>

--- a/cmd/minikube/cmd/delete.go
+++ b/cmd/minikube/cmd/delete.go
@@ -87,7 +87,7 @@ func (error DeletionError) Error() string {
 	return error.Err.Error()
 }
 
-var DeleteHostAndDirectoriesGetter = func(api libmachine.API, cc *config.ClusterConfig, profileName string) error {
+var hostAndDirsDeleter = func(api libmachine.API, cc *config.ClusterConfig, profileName string) error {
 	if err := killMountProcess(); err != nil {
 		out.FailureT("Failed to kill mount process: {{.error}}", out.V{"error": err})
 	}
@@ -300,7 +300,7 @@ func deleteProfile(ctx context.Context, profile *config.Profile) error {
 		}
 	}
 
-	if err := DeleteHostAndDirectoriesGetter(api, cc, profile.Name); err != nil {
+	if err := hostAndDirsDeleter(api, cc, profile.Name); err != nil {
 		return err
 	}
 

--- a/cmd/minikube/cmd/delete.go
+++ b/cmd/minikube/cmd/delete.go
@@ -87,6 +87,24 @@ func (error DeletionError) Error() string {
 	return error.Err.Error()
 }
 
+var DeleteHostAndDirectoriesGetter = func(api libmachine.API, cc *config.ClusterConfig, profileName string) error {
+	if err := killMountProcess(); err != nil {
+		out.FailureT("Failed to kill mount process: {{.error}}", out.V{"error": err})
+	}
+
+	deleteHosts(api, cc)
+
+	// In case DeleteHost didn't complete the job.
+	deleteProfileDirectory(profileName)
+	deleteMachineDirectories(cc)
+
+	if err := deleteConfig(profileName); err != nil {
+		return err
+	}
+
+	return deleteContext(profileName)
+}
+
 func init() {
 	deleteCmd.Flags().BoolVar(&deleteAll, "all", false, "Set flag to delete all profiles")
 	deleteCmd.Flags().BoolVar(&purge, "purge", false, "Set this flag to delete the '.minikube' folder from your user directory.")
@@ -282,23 +300,10 @@ func deleteProfile(ctx context.Context, profile *config.Profile) error {
 		}
 	}
 
-	if err := killMountProcess(); err != nil {
-		out.FailureT("Failed to kill mount process: {{.error}}", out.V{"error": err})
-	}
-
-	deleteHosts(api, cc)
-
-	// In case DeleteHost didn't complete the job.
-	deleteProfileDirectory(profile.Name)
-	deleteMachineDirectories(cc)
-
-	if err := deleteConfig(profile.Name); err != nil {
+	if err := DeleteHostAndDirectoriesGetter(api, cc, profile.Name); err != nil {
 		return err
 	}
 
-	if err := deleteContext(profile.Name); err != nil {
-		return err
-	}
 	out.Step(style.Deleted, `Removed all traces of the "{{.name}}" cluster.`, out.V{"name": profile.Name})
 	return nil
 }

--- a/cmd/minikube/cmd/delete_test.go
+++ b/cmd/minikube/cmd/delete_test.go
@@ -117,6 +117,7 @@ func TestDeleteProfile(t *testing.T) {
 				t.Logf("load failure: %v", err)
 			}
 
+			DeleteHostAndDirectoriesGetter = DeleteHostAndDirectoriesMock
 			errs := DeleteProfiles([]*config.Profile{profile})
 			if len(errs) > 0 {
 				HandleDeletionErrors(errs)

--- a/cmd/minikube/cmd/delete_test.go
+++ b/cmd/minikube/cmd/delete_test.go
@@ -117,7 +117,7 @@ func TestDeleteProfile(t *testing.T) {
 				t.Logf("load failure: %v", err)
 			}
 
-			DeleteHostAndDirectoriesGetter = DeleteHostAndDirectoriesMock
+			hostAndDirsDeleter = hostAndDirsDeleterMock
 			errs := DeleteProfiles([]*config.Profile{profile})
 			if len(errs) > 0 {
 				HandleDeletionErrors(errs)
@@ -158,7 +158,7 @@ func TestDeleteProfile(t *testing.T) {
 	}
 }
 
-var DeleteHostAndDirectoriesMock = func(api libmachine.API, cc *config.ClusterConfig, profileName string) error {
+var hostAndDirsDeleterMock = func(api libmachine.API, cc *config.ClusterConfig, profileName string) error {
 	return deleteContextTest()
 }
 
@@ -222,7 +222,7 @@ func TestDeleteAllProfiles(t *testing.T) {
 	}
 
 	profiles := append(validProfiles, inValidProfiles...)
-	DeleteHostAndDirectoriesGetter = DeleteHostAndDirectoriesMock
+	hostAndDirsDeleter = hostAndDirsDeleterMock
 	errs := DeleteProfiles(profiles)
 
 	if errs != nil {

--- a/cmd/minikube/cmd/delete_test.go
+++ b/cmd/minikube/cmd/delete_test.go
@@ -17,15 +17,18 @@ limitations under the License.
 package cmd
 
 import (
+	"fmt"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"testing"
 
+	"github.com/docker/machine/libmachine"
 	"github.com/google/go-cmp/cmp"
 	"github.com/otiai10/copy"
 	"github.com/spf13/viper"
 
+	cmdcfg "k8s.io/minikube/cmd/minikube/cmd/config"
 	"k8s.io/minikube/pkg/minikube/config"
 	"k8s.io/minikube/pkg/minikube/localpath"
 )
@@ -154,6 +157,17 @@ func TestDeleteProfile(t *testing.T) {
 	}
 }
 
+var DeleteHostAndDirectoriesMock = func(api libmachine.API, cc *config.ClusterConfig, profileName string) error {
+	return deleteContextTest()
+}
+
+func deleteContextTest() error {
+	if err := cmdcfg.Unset(config.ProfileName); err != nil {
+		return DeletionError{Err: fmt.Errorf("unset minikube profile: %v", err), Errtype: Fatal}
+	}
+	return nil
+}
+
 func TestDeleteAllProfiles(t *testing.T) {
 	td, err := ioutil.TempDir("", "all")
 	if err != nil {
@@ -207,6 +221,7 @@ func TestDeleteAllProfiles(t *testing.T) {
 	}
 
 	profiles := append(validProfiles, inValidProfiles...)
+	DeleteHostAndDirectoriesGetter = DeleteHostAndDirectoriesMock
 	errs := DeleteProfiles(profiles)
 
 	if errs != nil {

